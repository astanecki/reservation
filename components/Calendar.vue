<template>
  <div class="calendar-wrapper">
    <span class="calendar-wrapper__dates-label">{{ labels.dates }}</span>
    <v-date-picker
      v-if="range.start && range.end"
      v-model="range"
      is-range
    >
      <template v-slot="{ inputValue, inputEvents }">
        <input
          :value="labels.checkIn"
          v-on="inputEvents.start"
        />
        <input
          :value="labels.checkOut"
          v-on="inputEvents.end"
        />
      </template>
    </v-date-picker>
  </div>
</template>
<script>
export default {
  props: {
    dateFrom: {
      type: String,
      required: true,
    },
    dateTo: {
      type: String,
      required: true,
    }
  },

  data() {
    return {
      labels: {
        // TODO i18n
        dates: 'Dates',
        checkIn: 'Check in',
        checkOut: 'Check out',
      },
      range: {
        start: '',
        end: '',
      },
    };
  },

  mounted() {
    this.range.start = new Date(this.dateFrom);
    this.range.end = new Date(this.dateTo);
  },
};
</script>
<style lang="scss" scoped>
.calendar-wrapper {
  display: flex;
  flex-direction: column;

  &__dates-label {
    text-align: left;
    font-weight: 700;
    color: #848484;
    font-size: 13px;
    margin-bottom: 5px;
  }
}
</style>
