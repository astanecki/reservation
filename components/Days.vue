<template>
  <div class="days">

  </div>
</template>
<script>
export default {
  name: 'Days',


}
</script>
<style lang="scss" scoped>
.days {

}
</style>
