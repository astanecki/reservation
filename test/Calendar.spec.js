import { mount } from "@vue/test-utils";
import { advanceTo, clear } from "jest-date-mock";

import Days from "~/components/Days";
import Calendar from "@/components/Calendar.vue";

describe("Calendar", () => {
  const requiredProps = {
    dateFrom: "2020-01-11",
    dateTo: "2021-01-11"
  };

  const createWrapper = (methods = {}) => {
    return mount(Calendar, {
      propsData: requiredProps,
      methods
    });
  };

  beforeEach(() => {
    advanceTo("2000-01-01");
  });

  afterEach(() => {
    clear();
  });

  it("is a Vue instance", () => {
    const wrapper = createWrapper();

    expect(wrapper.vm).toBeTruthy();
  });

  it(`should match initial snapshot`, () => {
    const wrapper = createWrapper();

    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  describe("data", () => {
    it(`should prepare basic data correctly`, () => {
      expect(Calendar.data()).toMatchInlineSnapshot(`
        Object {
          "month": "2000-01-01T00:00:00.000Z",
        }
      `);
    });
  });

  describe("events", () => {
    describe("startRangeChosen", () => {
      it(`should be called when startRangeChosen event fired on Days component`, () => {
        const wrapper = createWrapper();

        wrapper.vm.$emit = jest.fn();

        wrapper.findComponent(Days).vm.$emit("startRangeChosen");

        expect(wrapper.vm.$emit).toHaveBeenCalledWith('startRangeChosen');
      });
    });

    describe("endRangeChosen", () => {
      it(`should be called when endRangeChosen event fired on Days component`, () => {
        const mockedData = { start: '2020-01-20' };
        const wrapper = createWrapper();

        wrapper.vm.$emit = jest.fn();

        wrapper.findComponent(Days).vm.$emit("endRangeChosen", mockedData);

        expect(wrapper.vm.$emit).toHaveBeenCalledWith('endRangeChosen', mockedData);
      });
    });
  });
});
