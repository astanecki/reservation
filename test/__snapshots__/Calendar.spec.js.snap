// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should match initial snapshot 1`] = `
<div
  class="calendar-wrapper"
  datefrom="2020-01-11"
  dateto="2021-01-11"
>
  <div
    class="month-selector"
  >
    <span
      class="month-selector__arrow"
    >
      
    ‹
  
    </span>
     
    <span
      class="month-selector__header"
    >
      
    January 2000
  
    </span>
     
    <span
      class="month-selector__arrow"
    >
      
    ›
  
    </span>
  </div>
   
  <div
    class="days"
  >
    <div
      class="days__names"
    >
      <span>
        Sun
      </span>
      <span>
        Mon
      </span>
      <span>
        Tue
      </span>
      <span>
        Wed
      </span>
      <span>
        Thu
      </span>
      <span>
        Fri
      </span>
      <span>
        Sat
      </span>
    </div>
     
    <div
      class="days__list-of-days"
    >
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      26
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      27
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      28
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      29
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      30
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      31
    
      </span>
       
      <span
        class="days__list-of-days-item"
      >
        
      1
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      2
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      3
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      4
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      5
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--today"
      >
        
      6
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      7
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      8
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      9
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      10
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      11
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      12
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      13
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      14
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      15
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      16
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      17
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      18
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      19
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      20
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      21
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      22
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      23
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      24
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      25
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      26
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      27
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      28
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      29
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      30
    
      </span>
      <span
        class="days__list-of-days-item"
      >
        
      31
    
      </span>
       
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      1
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      2
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      3
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      4
    
      </span>
      <span
        class="days__list-of-days-item days__list-of-days-item--not-current-month"
      >
        
      5
    
      </span>
    </div>
  </div>
</div>
`;
